src/bots/depositor.py

```diff
@@ -0,0 +1,243 @@
+                message_schema=Schema(Or(DepositMessageSchema, PingMessageSchema)),
+            ))
+
+        if TransportType.KAFKA in variables.MESSAGE_TRANSPORTS:
+            transports.append(KafkaMessageProvider(
+                client=f'{variables.KAFKA_GROUP_PREFIX}deposit',
+                message_schema=Schema(Or(DepositMessageSchema, PingMessageSchema)),
+            ))
+
+        if not transports:
+            logger.warning({'msg': 'No transports found. Dry mode activated.', 'value': variables.MESSAGE_TRANSPORTS})
+
+        attest_prefix = self.w3.lido.deposit_security_module.get_attest_message_prefix()
+
+        self.message_storage = MessageStorage(
+            transports,
+            filters=[
+                message_metrics_filter,
+                get_deposit_messages_sign_filter(attest_prefix),
+            ],
+        )
+
+    def execute(self, block: BlockData) -> bool:
+        self._check_balance()
+
+        module_ids = self.w3.lido.staking_router.get_staking_module_ids()
+
+        for module_id in module_ids:
+            logger.info({'msg': f'Do deposit for module with id: {module_id}.'})
+            try:
+                self._deposit_to_module(module_id)
+            except ModuleNotSupportedError as error:
+                logger.warning({'msg': 'Module not supported exception.', 'error': str(error)})
+
+        return True
+
+    def _check_balance(self):
+        if variables.ACCOUNT:
+            balance = self.w3.eth.get_balance(variables.ACCOUNT.address)
+            ACCOUNT_BALANCE.set(balance)
+            logger.info({'msg': 'Check account balance.', 'value': balance})
+        else:
+            logger.info({'msg': 'No account provided. Dry mode.'})
+            ACCOUNT_BALANCE.set(0)
+
+    def _deposit_to_module(self, module_id: int) -> bool:
+        is_depositable = self._check_module_status(module_id)
+        logger.info({'msg': 'Fetch module depositable status.', 'value': is_depositable})
+
+        quorum = self._get_quorum(module_id)
+        logger.info({'msg': 'Build quorum.', 'value': quorum})
+
+        module_strategy = self._get_module_strategy(module_id)
+
+        gas_is_ok = module_strategy.is_gas_price_ok()
+        logger.info({'msg': 'Calculate gas recommendations.', 'value': gas_is_ok})
+
+        keys_amount_is_profitable = module_strategy.is_deposited_keys_amount_ok()
+        logger.info({'msg': 'Calculations deposit recommendations.', 'value': keys_amount_is_profitable})
+
+        if is_depositable and quorum and gas_is_ok and keys_amount_is_profitable:
+            logger.info({'msg': 'Checks passed. Prepare deposit tx.'})
+            return self._build_and_send_deposit_tx(quorum)
+
+        logger.info({'msg': 'Checks failed. Skip deposit.'})
+        return False
+
+    def _get_module_strategy(self, module_id: int) -> ModuleDepositStrategyInterface:
+        # ToDo somehow support different gas strategies for different module types
+        if module_id == 1:
```

Comment:
Maybe filter module by type here? You can use `StakingModule.getType()` to get module type information.

Reply:
Will be done in next PRs